src/main/frontend/src/pages/Registration.js

```diff
@@ -0,0 +1,97 @@
+import { useState } from "react";
+import {register} from "../utils/userUtils";
+import classes from "./Registration.module.css";
+import * as yup from 'yup';
+import { Formik, Form, Field } from 'formik';
+
+function Registration() {
+  const [registrationInfo, setRegistrationInfo] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+  });
+
+  const[responseState, setResponseState] = useState("");
+
+  const formValidation = yup.object().shape({
+    firstName: yup.string()
+        .required("*First name is required")
+        .matches(/^[^\p{P}\p{S}\s\d]*$/u,
+            "*Please remove any special characters"),
+    lastName: yup.string()
+        .required("*Last name is required")
+        .matches(/^([^\p{P}\p{S}\s\d]+[ -]?[^\p{P}\p{S}\s\d]+)*$/u,
+            "*Please remove any special characters"),
+    email: yup.string()
+        .email("*Email must be valid")
+        .required("*Email is required")
+        .matches( /^(([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5}){1,25})+([;.](([a-zA-Z0-9_\-\.]+)@{[a-zA-Z0-9_\-\.]+0\.([a-zA-Z]{2,5}){1,25})+)*$/,
+            "*Email is not valid"),
+            password: yup.string().required("*Password is required")
+            .matches(/^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/,
+                "*Password must be at least 8 characters long, and password must contain at least one digit, one lowercase and one uppercase letter!")
+});
+
+  const { firstName, lastName, email, password } = registrationInfo;
+
+  const changing = (e) =>
+  setRegistrationInfo({ ...registrationInfo, [e.target.name]: e.target.value });
+
+  const handleSubmit = async (user) => {
+    try {
+        console.log(user);
+        const person = await register(user);
+        setResponseState(person.response);
```

Comment:
After user is registered, we should either navigate to login page, or send login request to obtain token, and navigate to home